pycqed/instrument_drivers/meta_instrument/qubit_objects/qubit_object.py

```diff
@@ -421,26 +426,41 @@ def find_pulse_amplitude(self, amps,
             if n > max_n:
                 break
             else:
+                old_amp = ampl
                 ampl_span = 0.3*ampl/n
                 amps = np.linspace(ampl-ampl_span, ampl+ampl_span, 15)
                 self.measure_rabi(amps, n=n, MC=MC, analyze=False)
                 a = ma.Rabi_parabola_analysis(close_fig=close_fig)
+                # Decide which quadrature to take by comparing the contrast
                 if take_fit_I:
                     ampl = a.fit_res[0].params['x0'].value
+                elif (np.abs(max(a.fit_res[0].data) -
+                             min(a.fit_res[0].data))) > (
+                      np.abs(max(a.fit_res[1].data) -
+                             min(a.fit_res[1].data))):
+                    ampl = a.fit_res[0].params['x0'].value
                 else:
-                    if (a.fit_res[0].params['x0'].stderr <=
-                            a.fit_res[1].params['x0'].stderr):
+                    ampl = a.fit_res[1].params['x0'].value
+                if not min(amps) < ampl < max(amps):
+                    ampl_span *= 2
+                    amps = np.linspace(old_amp-ampl_span,
+                                       old_amp+ampl_span, 15)
+                    self.measure_rabi(amps, n=n, MC=MC, analyze=False)
+                    a = ma.Rabi_parabola_analysis(close_fig=close_fig)
+                    # Decide which quadrature to take by comparing the contrast
+                    if take_fit_I:
+                        ampl = a.fit_res[0].params['x0'].value
+                    elif (np.abs(max(a.fit_res[0].data) -
```

Comment:
Same comment above applies here, try to refer to the data directly instead of through the fit_res object. 

I think that in the future I want to also make this find amplitude compatible when only a single quadrature is acquired (as is the case with optimal weights). But that is beyond the scope of this PR.